api/src/main/java/org/pentaho/platform/api/genericfile/IGenericFileService.java

```diff
@@ -298,4 +298,22 @@ default IGenericFile getFile( @NonNull String path ) throws OperationFailedExcep
    * @throws OperationFailedException If the operation fails for some other (checked) reason.
    */
   void deleteFilesPermanently( @NonNull List<GenericFilePath> paths ) throws OperationFailedException;
+
+  /**
+   * Deletes files by sending them to the trash, given their paths.
+   *
+   * @param paths The list of file paths to be deleted.
+   * @throws AccessControlException   If the current user cannot perform this operation.
+   * @throws OperationFailedException If the operation fails for some other (checked) reason.
+   */
+  void deleteFiles( @NonNull List<GenericFilePath> paths ) throws OperationFailedException;
```

Comment:
Would be great to offer per-file operations as well. Would also make it easier to document the exceptions that can happen for each file.
Here, we're missing an explicit `@throws` for the `BatchOperationFailedException`, including an explanation of how the batch operation works with respect to individual exceptions.